src/goto-programs/slice_global_inits.cpp

```diff
@@ -32,8 +33,8 @@ void slice_global_inits(goto_modelt &goto_model)
   const irep_idt entry_point=goto_functionst::entry_point();
   goto_functionst &goto_functions=goto_model.goto_functions;
 
-  if(!goto_functions.function_map.count(entry_point))
-    throw "entry point not found";
+  INVARIANT(
+    goto_functions.function_map.count(entry_point), "entry point not found");
```

Comment:
Do all current invocations of this function make sure this invariant truly holds?

Reply:
Well it's not guaranteed by the invocation; I guess it's more like a property of a well-formed `goto-program`. Maybe a `DATA_INVARIANT` would be more appropriate.

Reply:
I'd say the fix is to make sure it is guaranteed by all current invocations. The `INVARIANT` then makes this future-proof as additional invocation points added later on may fail to do the same check.

Reply:
I dag a bit deeper and the input `goto_model` is (so far always) a product of `read_goto_binary`. I've added a post-condition to the `read_bin_goto_object_v4` function, so that only models with entry points are presently produced.

Reply:
Ok. I think we use `invalid_user_input` exceptions for invalid command line options. On one hand the error originates from the source code but on the other hand the well-formedness of a goto-programs requires an entry point. For now the code throws `incorrect_goto_program` exception, but I can re-write it, if `incorrect_source_program` would be more accurate.